File: src/main/java/org/funivan/lologa/tiles/Filter/ColorFilter.java
package org.funivan.lologa.tiles.Filter;

import org.funivan.lologa.tile.TileInterface;

import java.awt.*;

public class ColorFilter implements TileFilterInterface{
    private Color color;

    public ColorFilter(Color color) {
        this.color = color;
    }

    @Override
    public Boolean apply(TileInterface tileInterface) {
        return this.color.equals(tileInterface.color());
    }
}

File: src/main/java/org/funivan/lologa/tiles/Filter/ColorFilter.kt
package org.funivan.lologa.tiles.Filter

import org.funivan.lologa.tile.TileInterface
import java.awt.Color

class ColorFilter(private val color: Color) : TileFilterInterface {

    override fun apply(tileInterface: TileInterface): Boolean? {
        return this.color == tileInterface.color()
    }
}
